### PR TITLE
Continue to consolidate chef_object api to now include update

### PR DESCRIPTION
**Pull request by oferrigni**
chef_object:update is implemented for basic cases across all 
chef_objects.  More complex implemenations are still handled
by chef_db/chef_sql.

### CODE REVIEW

File: src/chef_object.erl [hunk 1]
```diff
@@ -34,6 +34,11 @@
                                  ReturnTransform :: tuple()}) ->
                                       select_return()).
 
+-type update_return() :: {ok, 1} | {ok, not_found} | {error, _}.
```
Comment (seth) [inline src/chef_object.erl:37]: I was sort of expecting we'd just reuse the select_callback and select_return (though I guess the return is different?)

File: src/chef_object.erl [hunk 2]
```diff
@@ -198,6 +207,11 @@ fetch(Rec, CallbackFun) ->
     Mod = element(1, Rec),
     Mod:fetch(Rec, CallbackFun).
 
+update(Rec, ActorId, CallbackFun) ->
+    Mod = element(1, Rec),
```
Comment (sdelano) [inline src/chef_object.erl:208]: We're inconsistently using both `element(1, Rec)` and and a helper `callback_mod(Rec)` for this.

**Comment by sdelano** [comment]
LGTM. This will make life much easier when we go to implement groups.

**Comment by seth** [comment]
:+1:

File: src/chef_object.erl (hunk 1 above)
Comment (seth) [inline src/chef_object.erl:37]: I'm pretty sure we'll run into a query that returns `{ok, N}` where `N > 1`.

Comment (seth) [inline src/chef_object.erl:37]: And so if we don't want to support that level of granularity, we should just say `ok`.

**Pull request merged**